packages/evolution-generator/src/tests/test_conditionals_generator.py

```
@@ -738,12 +738,41 @@ def test_invalid_comparison_operator_raises(self):
 
     def test_invalid_parentheses_raises(self):
-        """parentheses must be '(', ')', or empty."""
+        """parentheses must contain only valid opening/closing characters in order."""
         assert self.checker._validate_conditionals_row(
-            self._row(parentheses="(("), self.ROW_NUMBER
+            self._row(parentheses="(x"), self.ROW_NUMBER
         ) == [
             "Error in Conditionals sheet - Invalid parentheses in row 2: "
-            "must be one of ['(', ')'] or empty, got '(('"
+            "must contain only '(' and ')' characters with openings before closings, "
+            "or be empty, got '(x'"
```

Ruff (0.16.0)

[warning] Unparenthesized implicit string concatenation in collection

Did you forget a comma?

(ISC004)